joomla/joomla-cms
# Add locked field to extensions table (and use it in core extensions)

### Summary of Changes

This adds a new column to extensions table named `locked`.

This column is a flag 1 or 0 to check if extension can be uninstalled.

This PR also marks all core extensions as uninstallable (ie locked), but most are unprotected now so they can be disabled.

### Testing Instructions

First do a code review.

Two test needed: update, install

- Update
  - Apply patch
  - Run the update query manually 
  - Check in extensions manage if protected extensions are protected and locked extensions are marked as so
  - Check you cannot unninstall a locked extension but you can disable it if it's not protected.

- Install
  - Apply patch
  - Delete configuration.php and install joomla as usual
  - Check in extensions manage if protected extensions are protected and locked extensions are marked as so
  - Check you cannot unninstall a locked extension but you can disable it if it's not protected.

This needs to be tested in all 3 db systems.

### Documentation Changes Required

None.
